package/build-package.sh

```diff
@@ -3,32 +3,45 @@
 # This script builds a .deb package from the binaries in the .stack-work
```

Comment:
Can you run [shellcheck](https://github.com/koalaman/shellcheck) on this script?

If you want, you can add it to `default.nix` as `pkgs.shellcheck`

Reply:
> Can you run shellcheck on this script?

```
In package/build-package.sh line 12:
  echo '$VERSION must be set to build a package.'
       ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.
```
This is intentional, but I’ll remove the $ to avoid the warning.

```
In package/build-package.sh line 39:
export EXAMPLE_CONFIG_SHA256=$(sha256sum example-config.json | cut --characters=1-64)
       ^-------------------^ SC2155: Declare and assign separately to avoid masking return values.
```
Done.

```
In package/build-package.sh line 43:
cat deb-control  | envsubst > "$PKGNAME/DEBIAN/control"
    ^---------^ SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.

In package/build-package.sh line 44:
cat deb-postinst | envsubst > "$PKGNAME/DEBIAN/postinst"
    ^----------^ SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
```
Replaced with the `<` form. (`envstubst` does not accept a file name as argument.)

> you can add it to default.nix as pkgs.shellcheck

Great idea, added. Will also add a check to CI.

Reply:
> This is intentional, but I’ll remove the $ to avoid the warning.

FYI, shellcheck also has a feature to disable individual warnings using comments: https://github.com/koalaman/shellcheck/wiki/Ignore